### DISCUSSION TITLE
Archiving instead of allowing addons in the marketplace to be fully deleted. Thoughts?

### DISCUSSION THREAD

**Discussion by rickmills** [statamic/cms#4472] [Q&A] [answered]
Bit of a random one this.

We're in the middle of a site build and were using the 'Mapbox' addon by @jimblue, but the site's not launched yet. I was pulled in to help out with a couple of bits on the project, went to set it up locally to have composer error out saying it couldn't find the Mapbox repo.

For whatever reason it's been made private - that's obviously not Statamic's problem to deal with, and I'm sure theres a good reason for it.

What was troubling is that it's page on the Statamic marketplace is completely gone, and now 404's. I'm assuming addon devs have the ability to delete the addon from the site.

My suggestion would be that instead of allowing that to happen, allow developers to archive the addon, with an optional box to explain why. That way they can either say 'dont use this its no longer developed / insecure / etc' or 'I've stopped working on this, but check out this alternative addon'.

It did also raise the question of how are premium addons handled for stuff like this? Can an addon you've paid for be deleted, leaving you in the lurch? I know the addons files themselves are not distributed by Statamic, but it's potentially something that could cause a lot of problems for people. Maybe an archive of each release is needed.

Any thoughts / suggestions / criticism is welcome, I'm just opening this up for discussion about best solutions to improve the experience here a bit.

**Comment by jasonvarga** [comment, marked as answer]
This is a bit of an unusual situation. The product is not only gone from the marketplace - _all_ of Jim's repositories are gone from GitHub. Like you said, I'm sure he has his reasons for doing that.

We don't allow products to be deleted yet but we do allow you to set them as a draft which would hide them from the site until you're ready to go live. Also, if the associated GitHub repo disappears, it would disappear from the Marketplace since we use GitHub to drive a lot of the information there.

This isn't really a situation we had planned on to be honest. We have ways for sellers to mark products as unsupported and to change the price to free. Didn't expect people to just disappear from the internet.

If you still have `jimblue/mapbox` in your `vendor` directory, you could throw that on GitHub.

**Reply by jimblue** [reply]
@jasonvarga I do not allow copying / use of my work for free so I do not agree when you say:

> If you still have jimblue/mapbox in your vendor directory, you could throw that on GitHub.

@rickmills I'm sorry for the inconvenience but did you purchased any Mapbox add-on license ?

In fact, only 2 people did since I launched it, which is way much less than the total number of downloads over 10 months.

Obviously my goal was not to block any projects base on Mapbox add-on. But given the time taken to create and maintain this add-on, it wasn't acceptable anymore to me. As sad as it may be for me, I had to decide to make the repository private while still providing support to the only 2 buyers.

Still, I continue to develop and improve the Mapbox add-on so @rickmills, if you had plan to purchase a Mapbox add-on license before your site goes live, please email me at mr.jimblue@gmail.com so we can find a solution for your project.

**Reply by rickmills** [reply]
We'd certainly have been purchasing it, we wouldn't dream of using it on a live site without a valid license. The project it was used on is still in development and hasn't gone live yet. I'll pass this on to the main project team and get them to drop you an email if they plan to stick to using mapbox on the site as I know they started looking at other options given it looked like there was no alternative.

**Reply by jasonvarga** [reply]
> I do not allow copying / use of my work for free so I do not agree

I meant for him to put in a private repo so he can continue to use it since the source has disappeared. Not distribute it.

**Comment by jimblue** [comment]
To answer the following part of your message @rickmills :
> Any thoughts / suggestions / criticism is welcome, I'm just opening this up for discussion about best solutions to improve the experience here a bit.

I would say if one day the Statamic Marketplace offered a way to completely host and protect premium add-ons, I would be happy to hand over my Mapbox add-on.

This would ensure developers get paid for their work and buyers always get access to their purchases. The 25% commisions Statamic make on the Marketplace should allow that.

**Reply by jackmcdade** [reply]
Hey there Jim, I'm sorry you had a rough go of supporting your addons, that bums me out.

I just wanted to make sure you got a response from me on all the important points that came up in this discussion.

First, the number of downloads or installs is not the same thing as unique sites running an addon. Composer tracks each install and update, _on each machine_, as an install or download. Statamic has had [290,032 installs](https://packagist.org/packages/statamic/cms) but we've only sold about 1% that many (with 1/3 of them being v2 upgrades). 

Another thing to mention is that we allow and encourage developers to install addons and try them out before buying them. As per our terms, licenses aren't required to be purchased until a project has launched. That can also lead to installs being higher than purchases.

Next, it kinda feels like you're suggesting that there's piracy going on (my apologies if I misread it), but I really don't think that's the case. Every commercial addon reports itself as needing a license in the Control Panel if one hasn't been purchased and assigned to the site. It's annoying, and in your face, and it works. I'm sure there's always someone out there that hacks around this (and keeps hacking it over and over every time they update) but for the most part this system has worked very well for us over the last 10 years. We have an awesome community full of trustworthy folks.

Which brings me back to Composer again. Statamic addons are Composer packages, which are required to be open and public repositories. There really are no user friendly ways to build a modern PHP application without Composer. We certainly couldn't use Laravel without it. There are ways of hosting a private Composer instance, but the complexity goes WAY UP for the end user, and they have to authenticate secondary repositories with API tokens and we feel it would be an overwhelmingly negative experience for most of our users and get in the way of the product itself.

I'll say this too in case it helps — up until the last year (with v3), 3rd party addon sales were pretty low overall. Most addons only sold a few copies with the exception of SEO Pro, Akismet, and a few other utility ones. In the last year however, things have really started to change. 3rd party addons sales are up 73% over last year, which shows us that v3 and the Composer approach has **opened** open up the market, rather than the other way around.

I hope this info helps you or someone else!
